cql2pgjson/src/test/resources/foreignKey.sql

```diff
@@ -35,16 +39,31 @@ END; $$ language 'plpgsql';
 CREATE TRIGGER update_tablec_references BEFORE INSERT OR UPDATE ON tablec
   FOR EACH ROW EXECUTE PROCEDURE update_tablec_references();
 
+CREATE OR REPLACE FUNCTION update_tabled_references() RETURNS TRIGGER AS $$
+BEGIN
+  NEW.id       = NEW.jsonb->>'id';
+  
+  RETURN NEW;
+END; $$ language 'plpgsql';
+CREATE TRIGGER update_tabled_references BEFORE INSERT OR UPDATE ON tabled
+  FOR EACH ROW EXECUTE PROCEDURE update_tabled_references();
+INSERT INTO tabled (jsonb) VALUES
+('{"id": "D1111111-1111-1111-1111-111111111111", "prefix": "a1", "otherindex": "z1"}'),
+('{"id": "D2222222-2222-2222-2222-222222222222", "prefix": "a2", "otherindex": "z2"}'),
+('{"id": "D3333333-3333-3333-3333-333333333333", "prefix": "a2", "otherindex": "z3"}');  
+
 INSERT INTO tablea (jsonb) VALUES
-('{"id": "A0000000-0000-0000-0000-000000000000", "name": "test0"}'),
-('{"id": "A1111111-1111-1111-1111-111111111111", "name": "test1"}'),
-('{"id": "A2222222-2222-2222-2222-222222222222", "name": "test2"}'),
-('{"id": "A3333333-3333-3333-3333-333333333333", "name": "test3"}');
-INSERT INTO tablea (jsonb) VALUES (jsonb_build_object('id', md5('a' || generate_series(1, 2000)::text)));
+('{"id": "A0000000-0000-0000-0000-000000000000", "name": "test0","tabledId": "D1111111-1111-1111-1111-111111111111"}'),
+('{"id": "A1111111-1111-1111-1111-111111111111", "name": "test1","tabledId": "D1111111-1111-1111-1111-111111111111"}'),
+('{"id": "A2222222-2222-2222-2222-222222222222", "name": "test2","tabledId": "D2222222-2222-2222-2222-222222222222"}'),
+('{"id": "A3333333-3333-3333-3333-333333333333", "name": "test3","tabledId": "D3333333-3333-3333-3333-333333333333"}'),
+('{"id": "A4444444-4444-4444-4444-444444444444", "name": "test4"}');
 
 INSERT INTO tableb (jsonb) VALUES
 ('{"id": "B1111111-1111-1111-1111-111111111111", "prefix": "x1", "otherindex": "y1","tableaId": "A1111111-1111-1111-1111-111111111111"}'),
 ('{"id": "B2222222-2222-2222-2222-222222222222", "prefix": "x2", "otherindex": "y2", "tableaId": "A2222222-2222-2222-2222-222222222222"}'),
 ('{"id": "B3333333-3333-3333-3333-333333333333", "prefix": "x2", "otherindex": "y3","tableaId": "A2222222-2222-2222-2222-222222222222"}'),
 ('{"id": "B4444444-4444-4444-4444-444444444444", "prefix" : "x0'')));(((''DROP tableb","otherindex": "y4", "tableaId": "A3333333-3333-3333-3333-333333333333"}');
-INSERT INTO tableb (jsonb) VALUES (jsonb_build_object('id', md5('b' || generate_series(1, 2000)::text)));
```

Comment:
Using EXPLAIN ANALYSE shows what indexes postgresql uses. For that the additional records are needed, if there are only a few records then postgresql uses a full table scan.